src/from-promise.ts

```diff
@@ -78,7 +78,7 @@ function createObservablePromise(origPromise: any) {
     extendObservable(
         promise,
         {
-            value: undefined,
+            value: oldPromise ? oldPromise.value : undefined,
```

Comment:
I think we only want to grab that value if the status is `fullfilled`?

Reply:
something like this condition on the old value? 
` oldPromise.state === FULFILLED? oldPromise.value: undefined;`

Reply:
updated